### PR TITLE
Add Segment Anything Model guide to the KerasCV guides

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
This PR adds a guide for KerasCV's `SegmentAnythingModel` model released in v0.7.0. It covers point, box, and text prompts and also demonstrates a quick way to optimize the model. Text prompts are supported via the PyTorch implementation of the [GoundingDINO model](https://github.com/IDEA-Research/GroundingDINO).

@fchollet I also wanted to add the predictor and automatic mask generator demos at [my repo](https://github.com/tirthasheshpatel/segment_anything_keras). Right now, I just add a link to it but I will try to add all the required code to KerasCV and update the guide. I decided against pulling the code from my repo to avoid breaking the guide if there are any changes there.

### CODE REVIEW

File: scripts/guides_master.py [hunk 1]
```diff
@@ -27,6 +27,10 @@
             "path": "semantic_segmentation_deeplab_v3_plus",
             "title": "Semantic Segmentation with KerasCV",
         },
+        {
+            "path": "segment_anything_in_keras_cv",
+            "title": "Segment Anything in KerasCV",
```
Comment (fchollet) [inline scripts/guides_master.py:32]: You can also add the keras 3 tag.

File: guides/keras_cv/segment_anything_in_keras_cv.py [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+Voila! The model understood that the object we wanted to exclude from our mask was the
+rim of the tyre.
+
+## Text prompts
+
+Finally, let's see how text prompts can be used along with KerasCV's
+`SegmentAnythingModel`.
+
+For this demo, we will use the [offical grounding dino
+model](https://github.com/IDEA-Research/GroundingDINO). Grounding DINO is a model that
```
Comment (fchollet) [inline guides/keras_cv/segment_anything_in_keras_cv.py:359]: For markdown links to render properly, they need to be on a single line, unbroken. The `markdown` library doesn't support links like the above (GitHub and the like presumably use a private fork that paves over these issues)

File: guides/keras_cv/segment_anything_in_keras_cv.py [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+```
+pip install -U git+https://github.com/IDEA-Research/GroundingDINO.git
+```
+
+Then, we can install the pretrained model's weights and config:
+"""
+
+"""shell
+wget -q https://github.com/IDEA-Research/GroundingDINO/releases/download/v0.1.0-alpha/groundingdino_swint_ogc.pth
+wget -q https://raw.githubusercontent.com/IDEA-Research/GroundingDINO/main/groundingdino/config/GroundingDINO_SwinT_OGC.py
```
Comment (fchollet) [inline guides/keras_cv/segment_anything_in_keras_cv.py:377]: Pulling from `main` might break in the future, can you pull from the 0.1.0 release?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR.

This is great content!

File: scripts/guides_master.py (hunk 1 above)
Comment (tirthasheshpatel) [reply scripts/guides_master.py:32]: I think all the guides under `guides/` are supposed to work with Keras 3. We didn't add this tag when porting the guides to Keras 3. Also, I don't see the tag anywhere in the file. Do we need to add this tag to other guides too?

File: guides/keras_cv/segment_anything_in_keras_cv.py (hunk 2 above)
Comment (tirthasheshpatel) [reply guides/keras_cv/segment_anything_in_keras_cv.py]: Done, thanks!

File: guides/keras_cv/segment_anything_in_keras_cv.py (hunk 3 above)
Comment (tirthasheshpatel) [reply guides/keras_cv/segment_anything_in_keras_cv.py]: Done.

File: scripts/guides_master.py (hunk 1 above)
Comment (fchollet) [reply scripts/guides_master.py:32]: Ok, let's skip then!

File: guides/keras_cv/segment_anything_in_keras_cv.py [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+Voila! The model understood that the object we wanted to exclude from our mask was the
+rim of the tyre.
+
+## Text prompts
+
+Finally, let's see how text prompts can be used along with KerasCV's
+`SegmentAnythingModel`.
+
+For this demo, we will use the
+[offical grounding dino model](https://github.com/IDEA-Research/GroundingDINO).
```
Comment (pcoet) [inline guides/keras_cv/segment_anything_in_keras_cv.py:359]: Nit: "grounding dino" -> "Grounding DINO"

**Review by pcoet — APPROVED** [review]
Nice writing!

File: guides/keras_cv/segment_anything_in_keras_cv.py (hunk 4 above)
Comment (tirthasheshpatel) [reply guides/keras_cv/segment_anything_in_keras_cv.py]: Done, thanks!

**Review by fchollet — APPROVED** [review]
Awesome work! LGTM
